runtime/runtime/src/lib.rs

```diff
@@ -2378,6 +2446,47 @@ struct ApplyProcessingReceiptState<'a> {
     pipeline_manager: pipelining::ReceiptPreparationPipeline,
 }
 
+trait MaybeRefReceipt {
+    fn as_ref(&self) -> &Receipt;
+}
+
+impl MaybeRefReceipt for Receipt {
+    fn as_ref(&self) -> &Receipt {
+        self
+    }
+}
+
+impl<'a> MaybeRefReceipt for &'a Receipt {
+    fn as_ref(&self) -> &Receipt {
+        *self
+    }
+}
+
+/// Schedule a one receipt for contract preparation.
+///
+/// The caller should call this method again after the returned number of receipts are processed.
```

Comment:
nit: Can you also add what is the return value?

Reply:
I'm not sure what is it exactly that you're asking for? Are you asking to document that it is an `Option<usize>`? That's already documented (in the function signature.) The doc-comment itself also explains how to interpret the return value, so I'm really stumped as to what else I could add here :)

Reply:
Ah sorry I somehow didn't parse the second line of the comment fully.